src/app/InteractionModelEngine.cpp

```diff
@@ -205,34 +205,48 @@ CHIP_ERROR InteractionModelEngine::OnInvokeCommandRequest(Messaging::ExchangeCon
     return CHIP_NO_ERROR;
 }
 
-CHIP_ERROR InteractionModelEngine::OnReadInitialRequest(Messaging::ExchangeContext * apExchangeContext,
-                                                        const PayloadHeader & aPayloadHeader,
-                                                        System::PacketBufferHandle && aPayload,
-                                                        ReadHandler::InteractionType aInteractionType,
-                                                        Protocols::InteractionModel::Status & aStatus)
+CHIP_ERROR InteractionModelEngine::CheckWhetherKeepExistingSubscription(Messaging::ExchangeContext * apExchangeContext,
+                                                                        System::PacketBufferHandle && aPayload)
 {
-    ChipLogDetail(InteractionModel, "Received %s request",
-                  aInteractionType == ReadHandler::InteractionType::Subscribe ? "Subscribe" : "Read");
+    bool keepSubscriptions = true;
+    System::PacketBufferTLVReader reader;
+    reader.Init(aPayload.Retain());
+    ReturnErrorOnFailure(reader.Next());
+    SubscribeRequestMessage::Parser subscribeRequestParser;
+    ReturnErrorOnFailure(subscribeRequestParser.Init(reader));
+    CHIP_ERROR err = subscribeRequestParser.GetKeepSubscriptions(&keepSubscriptions);
+    if (CHIP_END_OF_TLV == err)
+    {
+        return CHIP_NO_ERROR;
+    }
+    else if (CHIP_END_OF_TLV != err && CHIP_NO_ERROR != err)
+    {
+        return err;
+    }
 
```

Comment:
How about:
```suggestion
    if (keepSubscriptions) 
    {
        return CHIP_NO_ERROR;
    }
    
```
and skip the loop over read handlers if we know it's going to be a no-op?  And then not do the test in that loop body.

Reply:
Done, thanks